### PR TITLE
fix: Hamiltonian ascii simplification

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Default ASCII representation of Hamiltonians (Sum observables) to "Hamiltonian".

Opt to underestimate qubit count for Sum observables rather than overestimate

*Testing done:*

unit testing, manual performance testing

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/circuits/observables.py [hunk 1]
```diff
@@ -431,7 +431,7 @@ def _compute_eigenvalues(observables: Tuple[Observable], num_qubits: int) -> np.
 class Sum(Observable):
     """Sum of observables"""
 
-    def __init__(self, observables: List[Observable]):
+    def __init__(self, observables: List[Observable], display_name: str = "Hamiltonian"):
```
Comment (christianbmadsen) [inline src/braket/circuits/observables.py:434]: Should'nt the display name be "Observable", "O" or "..."? Is it guaranteed to be a Hamiltonian?

Comment (ajberdy) [reply src/braket/circuits/observables.py:434]: Yes. This constructor is only called in our code when someone uses the `+` operator between two terms, so it is almost certainly a multi-term Hamiltonian. The edge case exception being if someone manually constructs a Sum observable with a list of length one, which would make it a single-term Hamiltonian (unlikely and still technically correct).

File: test/unit_tests/braket/circuits/test_observable.py [hunk 2]
```diff
@@ -212,5 +212,5 @@ def test_sum_observable_with_subtraction():
     obs2 = -4 * Observable.Y()
     result = obs1 - obs2
     assert isinstance(result, Observable.Sum)
-    assert result.qubit_count == 2
-    assert result.ascii_symbols == ("6X+4Y", "6X+4Y")
+    assert result.qubit_count == 1
+    assert result.ascii_symbols == ("6X+4Y",)
```
Comment (kshyatt-aws) [inline test/unit_tests/braket/circuits/test_observable.py:216]: Can we add a test (unit or integ) using an actual large Hamiltonian, so that if a similar bug recurs, the tests will fail/timeout?

**Review by christianbmadsen — APPROVED** [review]

File: test/unit_tests/braket/circuits/test_observable.py (hunk 2 above)
Comment (ajberdy) [reply test/unit_tests/braket/circuits/test_observable.py:216]: Hesitant to add purposefully long-running tests to our integ tests since that increases friction and costs. This would be a great candidate for continuous benchmarking and I believe we should add that as a follow-up

File: src/braket/circuits/observables.py (hunk 1 above)
Comment (rmshaffer) [reply src/braket/circuits/observables.py:434]: I think that whether it is a "Hamiltonian" or not is a problem-specific detail that is not known here? Often it will indeed be a Hamiltonian in the case where the observable is measuring energy. But in general the observable could be measuring anything.

Comment (ajberdy) [reply src/braket/circuits/observables.py:434]: I believe it's a Hamiltonian in the sense that it is literally the sum of observables being measured. That may directly translate to an energy Hamiltonian of a simulated system or some other problem-specific construct, but the actual quantity being measured is a Hamiltonian from the physical system created in the QPU. At least, that's my current understanding.
